dev-scripts/deploy_custom_rh-che.sh

```diff
@@ -441,13 +441,26 @@ if ! (echo "$CHE_APP_CONFIG_YAML" | oc process -f - | oc apply -f - > /dev/null
   exit 5
 fi
 
+
 # APPLY SECRET IF REQUESTED
 if [ "$RH_CHE_APPLY_SECRET" == "true" ]; then
   echo "$RH_CHE_OC_SECRET" | oc apply -f -
   oc secrets link default quay-dev-deployer --for=pull
   oc secrets link rhche quay-dev-deployer --for=pull
 fi
 
+# CHECK IF THE SECRET IS SET PROPERLY, RE-SET IF NEEDED
+sleep 3
+echo "checking if the pod has quay-dev-deployer secret"
+pod=$(oc get pod -l app=rhche -o name)
+secret_contain=$(oc get -o json ${pod} | jq .spec.imagePullSecrets[] | grep quay-dev-deployer )
+if [ ${#secret_contain} > 0 ]; then
+  echo "Secret is present in the rhche pod, there should not be problem with pulling images."
+else
+  echo "Secret is not present in the rhche pod, removing the pod"
+  oc delete $(oc get pod -l app=rhche -o name)
```

Comment:
should we delete pod or dc here? or the idea is that pod will be re-created automatically?

Reply:
Yes, the idea is just to re-create pod. From how I understand the flow, the dc should be configured fine.

Reply:
But it's hard to test that as this issue is happening just from time to time.

Reply:
since it is hard to verify properly, I would postpone merging the PR unless we would have issues with https://github.com/redhat-developer/rh-che/pull/2039

Reply:
@ibuziuk This should fix the issue. If the issue is not appearing, the workaround will just not be triggered. If the problem appears, it will redeploy the pod and fix the issue. I don't see any harm that could be caused by merging that now - it can only bring benefits if the issue will come back again.